
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>MutCount &#8212; AdaptSearch 2 documentation</title>
    <link rel="stylesheet" href="_static/classic.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '2',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true,
        SOURCELINK_SUFFIX: '.txt'
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="ConcatPhyl" href="ConcatPhyl.html" />
    <link rel="prev" title="CDS_Search" href="CDS_Search.html" /> 
  </head>
  <body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="ConcatPhyl.html" title="ConcatPhyl"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="CDS_Search.html" title="CDS_Search"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">AdaptSearch 2 documentation</a> &#187;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="mutcount">
<h1>MutCount<a class="headerlink" href="#mutcount" title="Permalink to this headline">¶</a></h1>
<div class="section" id="overview">
<h2>Overview<a class="headerlink" href="#overview" title="Permalink to this headline">¶</a></h2>
<p>This script counts the number of codons, amino acids, and types of amino acids in sequences, as well as the mutation bias from one item to another between 2 sequences. Counting is then compared to empirical p-values, obtained from bootstrapped sequences obtained from a subset of sequences.</p>
<p>In the output files, the pvalues indicate the position of the observed data in a distribution of empirical countings obtained from a resample of the data. Values above 0.95 indicate a significantly higher counting, values under 0.05 a significantly lower counting.</p>
<p>The script resamples random pairs of aligned codon to determine what countings can be expected under the hypothesis of an homogenous dataset. Countings are performed on each generated random alignement, thousands of alignments allow to draw a gaussian distribution of the countings. Then the script simply checks whether the observed data are within the 5% lowest or 5% highest values of the distribution.</p>
<div class="section" id="user-parameters">
<h3>User parameters<a class="headerlink" href="#user-parameters" title="Permalink to this headline">¶</a></h3>
<dl class="docutils">
<dt>Two modes available :</dt>
<dd><ol class="first last arabic simple">
<li>Separated</li>
<li>Concatenated</li>
</ol>
</dd>
</dl>
<div class="section" id="separated-mode">
<h4>Separated mode<a class="headerlink" href="#separated-mode" title="Permalink to this headline">¶</a></h4>
<blockquote>
<div><ol class="arabic simple">
<li>Inputs are all fasta files from <a class="reference external" href="CDS_Search.html">CDS_Search</a> separated by commas</li>
<li>Concatenated file from <a class="reference external" href="ConcatPhyl.html">ConcatPhyl</a> is used to retrieve all species names</li>
</ol>
</div></blockquote>
</div>
<div class="section" id="concatenated-mode">
<h4>Concatenated mode<a class="headerlink" href="#concatenated-mode" title="Permalink to this headline">¶</a></h4>
<blockquote>
<div><ol class="arabic simple">
<li>Input fasta file is the ‘Phylogeny_concatenation_fasta_nuc’ from <a class="reference external" href="ConcatPhyl.html">ConcatPhyl</a></li>
<li>List of species for countings : the user can focus countings on these species only</li>
<li>List of species used for resampling : the user can focus the resampling on these species only</li>
<li>Number of sampled codons : sets the length (in codons) of the resampled sequences</li>
<li>Number of iterations : sets the number of resampled sequences</li>
</ol>
</div></blockquote>
</div>
</div>
</div>
<div class="section" id="code-documentation">
<h2>Code documentation<a class="headerlink" href="#code-documentation" title="Permalink to this headline">¶</a></h2>
<div class="section" id="algorithm">
<h3>Algorithm<a class="headerlink" href="#algorithm" title="Permalink to this headline">¶</a></h3>
<div class="section" id="id2">
<h4>Separated mode :<a class="headerlink" href="#id2" title="Permalink to this headline">¶</a></h4>
<div class="admonition-todo admonition" id="index-0">
<p class="first admonition-title">Todo</p>
<p class="last">Algo !</p>
</div>
</div>
<div class="section" id="id3">
<h4>Concatenated mode :<a class="headerlink" href="#id3" title="Permalink to this headline">¶</a></h4>
<dl class="docutils">
<dt>A whole bunch of dictionaries are used to store, for each sequences/pairs of sequences :</dt>
<dd><ol class="first last arabic simple">
<li>Each codons/amino-acids/amino-acids types occurrences countings/frequences (3*2=6 dictionaries)</li>
<li>Each codons/amino-acids/amino-acids types transitions countings/frequences (3*2=6 dictionaries)</li>
</ol>
</dd>
</dl>
<div class="highlight-python"><div class="highlight"><pre><span></span><span class="c1"># Example</span>
<span class="n">codons_counts_species_X</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;aaa&#39;</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span> <span class="s1">&#39;aac&#39;</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span> <span class="o">...</span><span class="p">}</span>
<span class="n">codons_transitions_counts_species_X_to_species_Y</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;aaa&#39;</span><span class="p">:{</span><span class="s1">&#39;aaa&#39;</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span> <span class="s1">&#39;aac&#39;</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span> <span class="o">...</span><span class="p">},</span>
                                                    <span class="s1">&#39;aac&#39;</span><span class="p">:{</span><span class="s1">&#39;aaa&#39;</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span> <span class="s1">&#39;aac&#39;</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span> <span class="o">...</span><span class="p">},</span>
                                                    <span class="o">...</span><span class="p">}</span>
</pre></div>
</div>
<p>Resampling is done on all or on a subset of sequences by picking codons at random positions. The more resemaplin</p>
<p>Then, Final dictionaries gathers all occurrences and transitions of all sequences, which are written to csv files using pandas.</p>
<div class="highlight-python"><div class="highlight"><pre><span></span><span class="c1"># Example</span>
<span class="n">codons_counts_final</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;Species1&#39;</span><span class="p">:</span> <span class="p">{</span><span class="s1">&#39;aaa&#39;</span><span class="p">:</span><span class="mi">20</span><span class="p">,</span> <span class="s1">&#39;aac&#39;</span><span class="p">:</span><span class="mi">12</span><span class="p">,</span> <span class="o">...</span><span class="p">},</span>
                       <span class="s1">&#39;Species2&#39;</span><span class="p">:</span> <span class="p">{</span><span class="s1">&#39;aaa&#39;</span><span class="p">:</span><span class="mi">14</span><span class="p">,</span> <span class="s1">&#39;aac&#39;</span><span class="p">:</span><span class="mi">17</span><span class="p">,</span> <span class="o">...</span><span class="p">},</span>
                       <span class="o">...</span><span class="p">}</span>
<span class="n">codons__transitions_counts_final</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;species_X_to_species_Y&#39;</span><span class="p">:</span> <span class="p">{</span><span class="s1">&#39;aaa&#39;</span><span class="p">:{</span><span class="s1">&#39;aaa&#39;</span><span class="p">:</span><span class="mi">12</span><span class="p">,</span> <span class="s1">&#39;aac&#39;</span><span class="p">:</span><span class="mi">1</span><span class="p">,</span> <span class="o">...</span><span class="p">},</span>
                                                               <span class="s1">&#39;aac&#39;</span><span class="p">:{</span><span class="s1">&#39;aaa&#39;</span><span class="p">:</span><span class="mi">2</span><span class="p">,</span> <span class="s1">&#39;aac&#39;</span><span class="p">:</span><span class="mi">15</span><span class="p">,</span> <span class="o">...</span><span class="p">},</span>
                                                               <span class="o">...</span><span class="p">}</span>
                                    <span class="s1">&#39;species_X_to_species_Z&#39;</span><span class="p">:</span> <span class="p">{</span><span class="s1">&#39;aaa&#39;</span><span class="p">:{</span><span class="s1">&#39;aaa&#39;</span><span class="p">:</span><span class="mi">8</span><span class="p">,</span> <span class="s1">&#39;aac&#39;</span><span class="p">:</span><span class="mi">2</span><span class="p">,</span> <span class="o">...</span><span class="p">},</span>
                                                               <span class="s1">&#39;aac&#39;</span><span class="p">:{</span><span class="s1">&#39;aaa&#39;</span><span class="p">:</span><span class="mi">3</span><span class="p">,</span> <span class="s1">&#39;aac&#39;</span><span class="p">:</span><span class="mi">10</span><span class="p">,</span> <span class="o">...</span><span class="p">},</span>
                                                               <span class="o">...</span><span class="p">}</span>
                                    <span class="o">...</span><span class="p">}</span>
</pre></div>
</div>
<p>Various values (IVRYWEL, EKQH, PAYRESDGM, GC12 …) are stored in int/float variables, put in a list (one per species) and then also written in csv file.</p>
</div>
</div>
<div class="section" id="source-code">
<h3>Source code<a class="headerlink" href="#source-code" title="Permalink to this headline">¶</a></h3>
<dl class="function">
<dt id="buildDicts">
<code class="descname">buildDicts</code><span class="sig-paren">(</span><em>list_codons</em>, <em>content</em>, <em>dict_genetic_code</em>, <em>dict_aa_classif</em><span class="sig-paren">)</span><a class="headerlink" href="#buildDicts" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>list_codons</strong> (<em>list</em>) – all codons except stop-codons</li>
<li><strong>content</strong> (<em>int</em><em> (</em><em>for countings</em><em>) or </em><em>list</em><em> (</em><em>for resampling</em><em>)</em>) – specifies the keys type</li>
<li><strong>dict_genetic_code</strong> (<em>dict</em>) – Describe the genetic code (key : amino-acid, values : list of codons)</li>
<li><strong>dict_aa_classif</strong> (<em>dict</em>) – Describe the amino-acids classification (key : amino-acid type, values : list of amino-acids)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">6 dictionaries used for countings and frequencies storage on a species or resampling</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">dict</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<p>At each species or pair of species countings, a copy of the input dictionaries is done, and dictionaries for transitions are also built.</p>
<dl class="function">
<dt id="viable">
<code class="descname">viable</code><span class="sig-paren">(</span><em>seqs</em>, <em>pos</em>, <em>method</em><span class="sig-paren">)</span><a class="headerlink" href="#viable" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>seqs</strong> (<em>list</em>) – sequences, which must all have the same size</li>
<li><strong>pos</strong> (<em>int</em>) – positions, &lt;= len(seqs) -3</li>
<li><strong>method</strong> (<em>string</em>) – “all” or “any”</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">if the given position is viable in all the sequence</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">bool</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<p>Compute if, among a set of sequences, either at least one of the codons at the specified position has not a “-“, or not any codon has a “-“</p>
<dl class="function">
<dt id="computeAllCountingsAndFreqs">
<code class="descname">computeAllCountingsAndFreqs</code><span class="sig-paren">(</span><em>seq</em>, <em>list_codons</em>, <em>init_dict_codons</em>, <em>init_dict_aa</em>, <em>init_dict_classif</em>, <em>dict_genetic_code</em>, <em>dict_aa_classif</em><span class="sig-paren">)</span><a class="headerlink" href="#computeAllCountingsAndFreqs" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>seq</strong> (<em>string</em>) – a dna/rna sequence (a,t,c,g)</li>
<li><strong>list_codons</strong> (<em>list</em>) – all codons except stop-codons</li>
<li><strong>init_dict_codons</strong> (<em>dict</em>) – a copied output of <cite>buildDicts()</cite></li>
<li><strong>init_dict_aa</strong> (<em>dict</em>) – a copied output of <cite>buildDicts()</cite></li>
<li><strong>init_dict_classif</strong> (<em>dict</em>) – a copied output of <cite>buildDicts()</cite></li>
<li><strong>dict_genetic_code</strong> (<em>dict</em>) – Describe the genetic code (key : amino-acid, values : list of codons)</li>
<li><strong>dict_aa_classif</strong> (<em>dict</em>) – Describe the amino-acids classification (key : amino-acid type, values : list of amino-acids)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">6 variables : countings and frequencies of codons, amino-acid and amino-acid types</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">dict</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<p>Call all functions dedicated to the computation of occurences and frequencies of codons, amino-acids, amino-acids types</p>
<dl class="function">
<dt id="computeVarious">
<code class="descname">computeVarious</code><span class="sig-paren">(</span><em>seq</em>, <em>dict_aa_counts</em>, <em>dict_aa_types</em><span class="sig-paren">)</span><a class="headerlink" href="#computeVarious" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>seq</strong> (<em>string</em>) – a dna/rna sequence (a,t,c,g)</li>
<li><strong>dict_aa_counts</strong> (<em>dict</em>) – the countings of amino-acids on the param <cite>seq</cite></li>
<li><strong>dict_aa_types</strong> (<em>dict</em>) – the countings of amino-acids types on the param <cite>seq</cite></li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">GC3, GC12, IVYWREL, EKQH, PAYRESDGM, purineload, CvP values for <cite>seq</cite></p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">int and floats</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<p>Call al the functions for nucleic and amino-acids sequences description : <em>GC3, GC12, IVYWREL, EKQH, PAYRESDGM, purineload, CvP</em></p>
<dl class="function">
<dt id="computeAllBiases">
<code class="descname">computeAllBiases</code><span class="sig-paren">(</span><em>seq1</em>, <em>seq2</em>, <em>dico_codons_transi</em>, <em>dico_aa_transi</em>, <em>dico_aatypes_transi</em>, <em>reversecode</em>, <em>reverseclassif</em><span class="sig-paren">)</span><a class="headerlink" href="#computeAllBiases" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>seq1</strong> (<em>string</em>) – a dna/rna sequence (a,t,c,g)</li>
<li><strong>seq2</strong> (<em>string</em>) – a dna/rna sequence (a,t,c,g)</li>
<li><strong>dico_codons_transi</strong> (<em>dict of dicts</em>) – a copied output of <cite>buildDicts()</cite></li>
<li><strong>dico_aa_transi</strong> (<em>dict of dicts</em>) – a copied output of <cite>buildDicts()</cite></li>
<li><strong>dico_aatypes_transi</strong> (<em>dict of dicts</em>) – a copied output of <cite>buildDicts()</cite></li>
<li><strong>reversecode</strong> (<em>dict</em>) – the genetic_code, but reversed (key : codon, values : amino-acid)</li>
<li><strong>reverseclassif</strong> (<em>dict</em>) – the amino-acids classification, but reversed (key : amino-acid, values : amino-acid type)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">countings and frequencies of codons, amino-acid and amino-acid types transitions from <cite>seq1</cite> to <cite>seq2</cite></p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">dict of dicts</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<p>Compute all biases (transisitions codon-&gt;codon, aa-&gt;-aa, aa_type-&gt;aa_type) between two sequences</p>
<dl class="function">
<dt id="sampling">
<code class="descname">sampling</code><span class="sig-paren">(</span><em>dict_seq</em>, <em>nb_iter</em>, <em>len_sample</em>, <em>list_codons</em>, <em>genetic_code</em>, <em>aa_classif</em>, <em>reversecode</em>, <em>reverseclassif</em><span class="sig-paren">)</span><a class="headerlink" href="#sampling" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>dict_seq</strong> (<em>dict</em>) – contains the species name (keys) and sequences (values) used</li>
<li><strong>nb_iter</strong> (<em>int</em>) – the number of iterations</li>
<li><strong>len_sample</strong> (<em>int</em>) – the lenght (in codons) of each resampled sequence</li>
<li><strong>list_codons</strong> (<em>list</em>) – all codons except stop-codons</li>
<li><strong>genetic_code</strong> (<em>dict</em>) – Describe the genetic code (key : amino-acid, values : list of codons)</li>
<li><strong>aa_classif</strong> (<em>dict</em>) – Describe the amino-acids classification (key : amino-acid type, values : list of amino-acids)</li>
<li><strong>reversecode</strong> (<em>dict</em>) – the genetic_code, but reversed (key : codon, values : amino-acid)</li>
<li><strong>reverseclassif</strong> (<em>dict</em>) – the amino-acids classification, but reversed (key : amino-acid, values : amino-acid type)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">6 variables</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">dict and dict of dicts</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<p>Resample randomly codons from sequences (sort of bootsrap)</p>
<dl class="function">
<dt id="testPvalues">
<code class="descname">testPvalues</code><span class="sig-paren">(</span><em>dict_counts</em>, <em>dict_resampling</em>, <em>nb_iter</em><span class="sig-paren">)</span><a class="headerlink" href="#testPvalues" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>dict_counts</strong> (<em>dict</em>) – contains codons frequencies of a sequence, an output of <cite>computeAllCountingsAndFreqs()</cite></li>
<li><strong>dict_resampling</strong> (<em>dict</em>) – </li>
<li><strong>nb_iter</strong> (<em>int</em>) – the number of iterrations used for resampling</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">pvalues for each item (codon, amino-acid, amino-acid type countings or transitions)</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">dict</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<p>Computes where the observed value is located in the expected counting distribution</p>
<p>Back to <a class="reference external" href="index.html">main page</a>.</p>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h3><a href="index.html">Table Of Contents</a></h3>
<p class="caption"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="Filter_Assemblies.html">Filter_Assemblies</a></li>
<li class="toctree-l1"><a class="reference internal" href="Pairwise.html">Pairwise</a></li>
<li class="toctree-l1"><a class="reference internal" href="POGs.html">POGs</a></li>
<li class="toctree-l1"><a class="reference internal" href="BlastAlign.html">BlastAlign</a></li>
<li class="toctree-l1"><a class="reference internal" href="TransDecoder.html">TransDecoder</a></li>
<li class="toctree-l1"><a class="reference internal" href="OrthoFinder_OnlyGroups.html">OrthoFinder_OnlyGroups</a></li>
<li class="toctree-l1"><a class="reference internal" href="Orthogroups_Tool.html">Orthogroups_Tool</a></li>
<li class="toctree-l1"><a class="reference internal" href="CDS_Search.html">CDS_Search</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">MutCount</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#overview">Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="#code-documentation">Code documentation</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="ConcatPhyl.html">ConcatPhyl</a></li>
<li class="toctree-l1"><a class="reference internal" href="codeML.html">codeML</a></li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="CDS_Search.html"
                        title="previous chapter">CDS_Search</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="ConcatPhyl.html"
                        title="next chapter">ConcatPhyl</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/MutCount.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="ConcatPhyl.html" title="ConcatPhyl"
             >next</a> |</li>
        <li class="right" >
          <a href="CDS_Search.html" title="CDS_Search"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">AdaptSearch 2 documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2018, Victor Mataigne.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.6.7.
    </div>
  </body>
</html>